src/riverroute/RtmHistFile.F90

```diff
@@ -291,8 +291,9 @@ subroutine RtmHistHtapesBuild ()
     call get_prev_time(day, sec)
     do t=1,ntapes
        tape(t)%begtime = day + sec/secspday
+       locfnh(t) = ''
+       locfnhr(t) = ''
```

Comment:
I am not familiar with this code, so I may be misunderstanding things, but from what I can see, it looks like this routine (RtmHistHtapesBuild) is called _after_ RtmRestFileRead in initialization. It looks like RtmRestFileRead sets these variables, so I'm wondering if it's a problem for them to be blanked out here after they were read in through the restart. For example, does this cause problems in a run where you are appending to an existing history file from the previous iteration of the run?

It looks like CTSM handles this via a logical, `history_tape_in_use`, but that may be overkill here. Would it work right to initialize this variable where it's declared at the top of the module? Otherwise, I'm thinking we'd need a little pre-init routine for just these variables that is called near the start of Rtmini.

Reply:
I don't recall why these lines were added.   You can try removing them, but I would like the change in lines 1282-1289 merged.